### ISSUE TITLE
FNX3-16355 ⁃ [Bug] When opening a link in a new tab, pressing back should close the tab

### ISSUE THREAD

**Issue by buttercookie42** [mozilla-mobile/fenix#2483]
## Steps to reproduce
1. Open any page.
1. Open a link in a new tab and switch to it.
(3. Optionally navigate within the new tab)
(4. Press the device's back button until you reach the start of session history in the new tab)
1. Press back one more time.

### Expected behavior
The tab closes and I'm returned to the "parent" tab from which the new tab was originally opened.
This is how things currently work both in Fennec and also e.g. in Chrome, and basically the internal version of #2110.
Some related behaviour is also implemented for the tab list in Fennec, i.e. under some circumstances, closing a child tab from the tab list will then select the parent tab in the tab list (and scroll to it, sell also #1827) instead of simply selecting the next adjacent tab.

Also to note is that in Fennec the parent/child relationship persists across session restores, and also closing and restoring the child tab, as long as the original parent tab still exists.

### Actual behavior
I end up on the home screen.

### Device information
* Android device: Moto G4 Play, Android 6.0.1
* Fenix version: 1.0.9020, Build #11341807

**Comment by vesta0** [comment]
I believe this is intended behaviour. @topotropic please confirm.

**Comment by topotropic** [comment]
We didn't specify this. It makes sense to follow above described expected behavior.

**Comment by ekager** [comment]
This will require AC work because we don't have a concept of the "parent" tab exposed to us in Fenix

**Comment by ekager** [comment]
Opened AC issue: https://github.com/mozilla-mobile/android-components/issues/3256

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on the latest Nightly Build #12970607 from 24/10 using the following devices:
• Google Pixel 3a XL (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7)
• OnePlus A3 (Android 6.0.1)
• LG Nexus 4 (Android 5.1.1)

► Video
![20191024_144557](https://user-images.githubusercontent.com/51314259/67483150-d95b0580-f66d-11e9-8f33-80722971837f.gif)

**Comment by nahuhh** [comment]
While this was a much needed feature, I do think mixing the "expected behavior" with the old would be ideal. 

The original issue with workflow was mostly regarding expectations on the relationship between "parent" and "child" tabs. The new workflow, I feel, reaches too far and includes all previous tabs from all sessions. 

My suggestion for improvement below:
"Old" behaviour 
1. Press back til you reach the root of a tab and
2. The tab page would be shown
3. Press back again and
(This is where it goes bad)
4. The browser would closed.

New behavior is 
1. Press back til you reach the root of a tab and
2. The tab is closed and you are taken immediately to the *previous* tab
*note* should be restricted to parent tabs only. Old/nonrelated tabs shouldn't load automatically (but they do)

Which is the reason for this suggestion:
I think the good way to approach this is 

1. Press back til you reach the root of a tab and
2. The tab page should be shown (the current tab highlighted and should still be open)
3. Press back again and 
3a. If there is a parent tab: current tab closes and you are returned to the parent tab. 
3b. If there is no parent: the browser closes. 

Basically, I think it beneficial to make a stop at the "tab" page before killing the tab and returning to the parent tab.
